src/WebJobs.Script/FileProvisioning/PowerShell/PowerShellFileProvisioner.cs

```diff
@@ -30,22 +44,125 @@ public Task ProvisionFiles(string scriptRootPath)
 
         private void AddRequirementsFile(string scriptRootPath)
         {
-            string requirementsFilePath = Path.Combine(scriptRootPath, "requirements.psd1");
+            _logger.LogInformation($"Creating {RequirementsPsd1FileName}.");
+            string requirementsFilePath = Path.Combine(scriptRootPath, RequirementsPsd1FileName);
+
             if (!File.Exists(requirementsFilePath))
             {
-                string content = FileUtility.ReadResourceString($"Microsoft.Azure.WebJobs.Script.FileProvisioning.PowerShell.requirements.psd1");
-                File.WriteAllText(requirementsFilePath, content);
+                string requirementsContent = FileUtility.ReadResourceString($"Microsoft.Azure.WebJobs.Script.FileProvisioning.PowerShell.requirements.psd1");
+                string guidance = null;
+
+                try
+                {
+                    string majorVersion = GetLatestAzModuleMajorVersion();
+
+                    requirementsContent = requirementsContent.Replace("# 'Az'", "'Az'");
+                    requirementsContent = requirementsContent.Replace("MAJOR_VERSION", majorVersion);
+                }
+                catch
+                {
+                    guidance = "Uncomment the next line and replace the MAJOR_VERSION, e.g., 'Az' = '2.*'";
+                    _logger.LogWarning("Failed to get Az module version. Edit the requirements.psd1 file when the powershellgallery.com is accessible.");
```

Comment:
I understand the need for this change in core tools as we need enable offline development. Assuming this change is for portal create experience,
- Is this needed? Have we had customer cases where create app fails due to powershellgallery.com being unvailalbe?
- have you measured latency for operation: connecting to powershellgallery and fetching latest version? If the latency is significant this will increase the function app create time.

Reply:
Thanks @pragnagopa for input on this. Yes, this change is for the Portal create experience.

>Is this needed? Have we had customer cases where create app fails due to powershellgallery.com being unvailalbe?

Yes, this is needed to have a consistent experience across all clients. In addition, with this work, we no longer have to update requirements.psd1 with the latest major version for Az, which has a major release every 4-6 months. 

Yes, it is possible that the PSGallery could go down. We experience this at the beginning of the PowerShell Public Preview. 

> have you measured latency for operation: connecting to powershellgallery and fetching latest version? If the latency is significant this will increase the function app create time.

Yes, this will add an extra 1.3 seconds on average.

To measure the latency, I used the ```Measure-Command``` cmdlet from PowerShell + the Core Tools which already has the code that fetches the latest Az major version.

```
$maxNumberOfExecutions = 10
$executionTime = 0

1..$maxNumberOfExecutions | % {
    rm D:\MyFunctions\* -Recurse -Force
    $result = measure-command {func init --worker-runtime powershell}
    $executionTime += $result.TotalSeconds
}
"average time = $($executionTime/$maxNumberOfExecutions)"
```

Reply:
Hi @eamonoreilly, @anirudhgarg, @paulbatum,

This PR will fetch the latest Az major version progamatically when generating the requirements.psd1 file. As a result, this will add 1.3 seconds to the current flow when creating a **PowerShell function app** for in the Portal. Let's meet today to discuss this. Thanks.